src/main/java/com/microsoft/azure/batch/protocol/models/CloudServiceConfiguration.java

```diff
@@ -46,7 +46,7 @@
      * virtual machines have upgraded, currentOSVersion is updated to be the
      * same as targetOSVersion.
      */
-    @JsonProperty(value = "currentOSVersion")
+    @JsonProperty(value = "currentOSVersion", access = JsonProperty.Access.WRITE_ONLY)
```

Comment:
I don't know Java.... but shouldn't this be 'READ_ONLY'?

Reply:
WRITE_ONLY
Access setting that means that the property may only be written (set) for deserialization, but will not be read (get) on serialization, that is, the value of the property is not included in serialization.

Here the read/write is kind of different meaning as property read/write.

Reply:
aaaah I see :)